src/components/AccessibilityGuide/DynamicContent.stories.mdx

```diff
@@ -0,0 +1,204 @@
+should announce the entire content or only the changes. This is useful when an
+element's entire content is not replaced on every update but all of the content
+should still be read.
+
+- `aria-atomic="true"` - the entire content of the element will be read by the
+  assistive technology.
+- `aria-atomic="false"` - only the changes to the content will be read by the
+  assistive technology.
+
+For example, if the content of an element is updated to "Results found: 5" and
+then updated again to "Results found: 10", the assistive technology will announce
+"10" when `aria-atomic="false"` but will announce "Results found: 10" when
+`aria-atomic="true"`.
+
+```html
+<div class="dynamic-content-container" aria-live="polite" aria-atomic="true">
+  Results found:
+  <span class="dynamic-content">
+    <!-- Any dynamic content should be inserted, removed, or updated here. -->
+  </span>
+</div>
+```
+
+## Anti-Patterns
+
+**Do not dynamically add or remove the `aria-live` attribute from the element.**
+
+The `aria-live` attribute should be set on the element when the page is rendered.
+The following is **incorrect**:
+
+```html
+<div id="dynamic-content-container">
+  <!-- Any dynamic content should be inserted, removed, or updated here. -->
+</div>
+```
+
+```js
+const container = document.getElementById("dynamic-content-container");
+container.setAttribute("aria-live", "polite");
+```
+
+**Do not assign the `aria-live` attribute to an entire section or the
+entire page.**
+
+Add the `aria-live` attribute to the specific element that will be updated. For
+example, on a search results page that updates the amount of results found and
+the list of results, only add the `aria-live` attribute to the element that
+contains the number of results. The following is not recommended:
+
+```html
+<div id="search-results" aria-live="polite">
+  <div id="results-count">
+    <!-- Dynamic content. -->
+    <span>10 results found</span>
+  </div>
+  <ul id="results-list">
+    <!-- Dynamic list of li elements. -->
+    <li>
+      <p>Title: Result 1</p>
+      <p>Author: Author 1</p>
+    </li>
+    <li>
+      <p>Title: Result 2</p>
+      <p>Author: Author 2</p>
+    </li>
+    <!-- ... -->
+  </ul>
+</div>
+```
+
+This is not recommended because the screen reader will announce:
+
+```
+"10 results found Title: Result 1 Author: Author 1 Title: Result 2 Author: Author 2 ..."
+```
+
+This is information overload. Instead, add the `aria-live` attribute to the
+element that contains the updated number of results found. In this case, it's
+the element with `id` of `results-count` only.
+
+## NYPL Patterns
+
+### HelperErrorText
+
+The `HelperErrorText` component renders an empty `div` element with the
+`aria-live` attribute set to `"polite"` by default. This element can be used to
+display content that is dynamically updated through the `text` prop. This
+component should not be conditionally rendered on the page. Instead, the content
+should be dynamically updated.
```

Comment:
A bit shorter: The `HelperErrorText` component renders an empty `div` element with the `aria-live` attribute set to `"polite"` by default.  This component should _not_ be conditionally rendered on the page. Instead, the value of the `text` prop should be dynamically updated.